src/service/notification_service/handler.rs

```diff
@@ -0,0 +1,71 @@
+use super::{EventEnvelope, EventType, Result};
+use async_trait::async_trait;
+use log::info;
+
+/// Handle an event when we receive it from a channel.
+#[allow(dead_code)]
+#[async_trait]
+pub trait NotificationHandlerApi: Send + Sync {
+    /// Whether this handler handles the given event type.
+    fn handles_event(&self, event_type: &EventType) -> bool;
```

Comment:
I'm not sure I understand this part of the API. :thinking: 

The way I imagine it, is that there's several notification handlers and depending on the event, they might fire, or not. But couldn't this check just be within `handle_event`, that it checks if the given event is supported and if not, it returns Ok(()) and it's basically a no-op?

I feel like that would be a simpler API, but maybe I'm missing something where `handles_event` is used to execute (or not execute) additional custom logic in some cases.

Reply:
Idea here is that there will be some Stream consumer running that listens for 'push notifications'. That one will have a Vec of NotificationHandlers attached. When the stream receives a message it will look into the Envelope to determine the EventType and the trigger all handlers that are subscribed to this type (handles_event?). Essentially the event type acts as sort of a topic.

Reply:
OK got it, thanks. So I think the only difference would be:

```rust
for handler in handlers {
  if handler.handles_event(&event) {
    handler.handle_event(&event);
  }
}
```

vs.

```rust
for handler in handlers {
    handler.handle_event(&event);
}
```

where the check is moved inside `handle_event`. I guess there's an argument on both sides - one keeps the public API smaller, the other makes it explicit, that not every handler can handle every event.
I think both are fine :+1: